pkg/proxy/service_test.go

```diff
@@ -0,0 +1,534 @@
+				makeServicePortName("somewhere-else", "cluster-ip", "something"):     makeTestServiceInfo("172.16.55.4", 1234, "UDP", 0),
+				makeServicePortName("somewhere-else", "cluster-ip", "somethingelse"): makeTestServiceInfo("172.16.55.4", 1235, "UDP", 0),
+			},
+		},
+		{
+			// Case[4]: nodeport service
+			service: makeTestService("somewhere-else", "node-port", func(svc *api.Service) {
+				svc.Spec.Type = api.ServiceTypeNodePort
+				svc.Spec.ClusterIP = "172.16.55.10"
+				svc.Spec.Ports = addTestPort(svc.Spec.Ports, "blahblah", "UDP", 345, 678, 0)
+				svc.Spec.Ports = addTestPort(svc.Spec.Ports, "moreblahblah", "TCP", 344, 677, 0)
+			}),
+			expected: map[ServicePortName]*fakeServiceInfo{
+				makeServicePortName("somewhere-else", "node-port", "blahblah"):     makeTestServiceInfo("172.16.55.10", 345, "UDP", 0),
+				makeServicePortName("somewhere-else", "node-port", "moreblahblah"): makeTestServiceInfo("172.16.55.10", 344, "TCP", 0),
+			},
+		},
+		{
+			// Case[5]: load balancer service
+			service: makeTestService("somewhere", "load-balancer", func(svc *api.Service) {
+				svc.Spec.Type = api.ServiceTypeLoadBalancer
+				svc.Spec.ClusterIP = "172.16.55.11"
+				svc.Spec.LoadBalancerIP = "5.6.7.8"
+				svc.Spec.Ports = addTestPort(svc.Spec.Ports, "foobar", "UDP", 8675, 30061, 7000)
+				svc.Spec.Ports = addTestPort(svc.Spec.Ports, "baz", "UDP", 8676, 30062, 7001)
+				svc.Status.LoadBalancer = api.LoadBalancerStatus{
+					Ingress: []api.LoadBalancerIngress{
+						{IP: "10.1.2.4"},
+					},
+				}
+			}),
+			expected: map[ServicePortName]*fakeServiceInfo{
+				makeServicePortName("somewhere", "load-balancer", "foobar"): makeTestServiceInfo("172.16.55.11", 8675, "UDP", 0),
+				makeServicePortName("somewhere", "load-balancer", "baz"):    makeTestServiceInfo("172.16.55.11", 8676, "UDP", 0),
+			},
+		},
+		{
+			// Case[6]: load balancer service with only local traffic policy
+			service: makeTestService("somewhere", "only-local-load-balancer", func(svc *api.Service) {
+				svc.Spec.Type = api.ServiceTypeLoadBalancer
+				svc.Spec.ClusterIP = "172.16.55.12"
+				svc.Spec.LoadBalancerIP = "5.6.7.8"
+				svc.Spec.Ports = addTestPort(svc.Spec.Ports, "foobar2", "UDP", 8677, 30063, 7002)
+				svc.Spec.Ports = addTestPort(svc.Spec.Ports, "baz", "UDP", 8678, 30064, 7003)
+				svc.Status.LoadBalancer = api.LoadBalancerStatus{
+					Ingress: []api.LoadBalancerIngress{
+						{IP: "10.1.2.3"},
+					},
+				}
+				svc.Spec.ExternalTrafficPolicy = api.ServiceExternalTrafficPolicyTypeLocal
+				svc.Spec.HealthCheckNodePort = 345
+			}),
+			expected: map[ServicePortName]*fakeServiceInfo{
+				makeServicePortName("somewhere", "only-local-load-balancer", "foobar2"): makeTestServiceInfo("172.16.55.12", 8677, "UDP", 345),
+				makeServicePortName("somewhere", "only-local-load-balancer", "baz"):     makeTestServiceInfo("172.16.55.12", 8678, "UDP", 345),
+			},
+		},
+		{
+			// Case[7]: externalname service
```

Comment:
Nit: "externalname" should be "external name"

Reply:
nice finding! It's fixed now. PTAL. THANKS!